Mathlib/Algebra/MonoidAlgebra/Basic.lean

```diff
@@ -922,6 +922,31 @@ theorem domCongr_toAlgHom (e : G ≃* H) : (domCongr k A e).toAlgHom = mapDomain
 
 end lift
 
+section Linear
+
+variable {A : Type*} [Semiring k] [Semiring A] [Module k A]
+
+variable (k A) in
+/-- Interpret `single i` as a linear map (but with `MonoidAlgebra` in the type, rather than
+`Finsupp`). -/
+def lsingle (i : G) : A →ₗ[k] MonoidAlgebra A G :=
+  Finsupp.lsingle i
+
+lemma lsingle_def (i : G) :
+    lsingle k A i = Finsupp.lsingle i := rfl
```

Comment:
This lemma breaks the MonoidAlgebra / Finsupp abstraction, so probably shouldn't exist

Reply:
I needed to use linearity of `single`, and then, afterwards, properties of the multiplication on `A[G]` in e.g. `MonoidAlgebra.instBialgebra`. And if I use `Finsupp.lsingle, Finsupp.lhom_ext` then the monoid algebra gets unfolded to `Finsupp` too early; adding `MonoidAlgebra.lsingle` was the best way I could think of to get around this